### PR TITLE
split-wait-for-availability

### PR DESCRIPTION
**Pull request by leoporoli**

### CODE REVIEW

File: docs/lib-documentation.md [hunk 1]
```diff
@@ -120,17 +120,29 @@ Repartitions the network so that the connections between services match the spec
...
+
+* `serviceId`: The ID of the service to check.
+* `port`: The port (e.g. 8080) of the endpoint to check.
+* `path`: The path of the service to check, which must not start with a slash (e.g. `service/health`).
+* `initialDelayMilliseconds`: Number of milliseconds to wait until executing the first HTTP call
+* `retries`: Max number of HTTP call attempts that this will execute until giving up and returning an error
+* `retriesDelayMilliseconds`: Number of milliseconds to wait between retries
+* `bodyText`: If this value is non-empty, the endpoint will not be marked as available until this value is returned (e.g. `Hello World`). If this value is emptystring, no body text comparison will be done.
+
+### waitForEndpointAvailabilityHttpPost(ServiceID serviceId, uint32 port, String path, String requestBody, uint32 initialDelayMilliseconds, uint32 retries, uint32 retriesDelayMilliseconds, String bodyText)
```
Comment (mieubrisse) [inline docs/lib-documentation.md:136]: s/http get/HTTP POST/g

File: docs/lib-documentation.md [hunk 2]
```diff
@@ -120,17 +120,29 @@ Repartitions the network so that the connections between services match the spec
 * `partitionConnections`: Definitions of the connection state between the new partitions. If a connection between two partitions isn't defined in this map, the default connection will be used. Connections are not directional, so an error will be thrown if the same connection is defined twice (e.g. `Map[A][B] = someConnectionInfo`, and `Map[B][A] = otherConnectionInfo`).
 * `defaultConnection`: The network state between two partitions that will be used if the connection isn't defined in the partition connections map.
 
-### waitForEndpointAvailability(ServiceID serviceId, String httpMethod, uint32 port, String path, String requestBody, uint32 initialDelaySeconds, uint32 retries, uint32 retriesDelayMilliseconds, String bodyText)
-Waits until a service endpoint is available by making requests to the endpoint using the given parameters. An error is thrown if the number of retries is exceeded.
+### waitForEndpointAvailabilityHttpGet(ServiceID serviceId, uint32 port, String path, String requestBody, uint32 initialDelayMilliseconds, uint32 retries, uint32 retriesDelayMilliseconds, String bodyText)
```
Comment (mieubrisse) [inline docs/lib-documentation.md:123]: s/http post/HTTP GET/g

File: docs/lib-documentation.md [hunk 3]
```diff
@@ -120,17 +120,29 @@ Repartitions the network so that the connections between services match the spec
...
+Waits until a service endpoint is available by making requests to the endpoint using the given parameters, and the http post method. An error is thrown if the number of retries is exceeded.
+
+**Args**
+
+* `serviceId`: The ID of the service to check.
+* `port`: The port (e.g. 8080) of the endpoint to check.
+* `path`: The path of the service to check, which must not start with a slash (e.g. `service/health`).
+* `initialDelayMilliseconds`: Number of milliseconds to wait until executing the first HTTP call
```
Comment (mieubrisse) [inline docs/lib-documentation.md:131]: I like that you standardized this to millis - just make sure to adapt appropriately inside Kurt Core

File: docs/changelog.md [hunk 4]
```diff
@@ -1,4 +1,8 @@
 # TBD
+### Breaking Changes
+* Split method `NetworkContext.WaitForAvailability` to `NetworkContext.WaitForAvailabilityHttpGet` and `NetworkContext.WaitForAvailabilityHttpPost`
```
Comment (mieubrisse) [inline docs/changelog.md:3]: Comment 1: technically this is `WaitForEndpointAvailability` (this update missing the `Endpoint`)

Comment 2: Can we flip the ordering here, to be `WaitForHttpGetEndpointAvailability` and `WaitForHttpPostEndpointAvailability`? It'll read a bit more naturally

File: docs/lib-documentation.md [hunk 5]
```diff
@@ -120,17 +120,29 @@ Repartitions the network so that the connections between services match the spec
...
 * `defaultConnection`: The network state between two partitions that will be used if the connection isn't defined in the partition connections map.
 
-### waitForEndpointAvailability(ServiceID serviceId, String httpMethod, uint32 port, String path, String requestBody, uint32 initialDelaySeconds, uint32 retries, uint32 retriesDelayMilliseconds, String bodyText)
-Waits until a service endpoint is available by making requests to the endpoint using the given parameters. An error is thrown if the number of retries is exceeded.
+### waitForEndpointAvailabilityHttpGet(ServiceID serviceId, uint32 port, String path, String requestBody, uint32 initialDelayMilliseconds, uint32 retries, uint32 retriesDelayMilliseconds, String bodyText)
+Waits until a service endpoint is available by making requests to the endpoint using the given parameters, and the http post method. An error is thrown if the number of retries is exceeded.
```
Comment (mieubrisse) [inline docs/lib-documentation.md:124]: s/http post/HTTP GET/g

File: docs/lib-documentation.md (hunk 2 above)
Comment (mieubrisse) [inline docs/lib-documentation.md:123]: Same comment - can we rename this to `waitForHttpGetEndpointAvailability`, to read a bit more naturally?

File: docs/lib-documentation.md [hunk 6]
```diff
@@ -120,17 +120,29 @@ Repartitions the network so that the connections between services match the spec
...
+* `serviceId`: The ID of the service to check.
+* `port`: The port (e.g. 8080) of the endpoint to check.
+* `path`: The path of the service to check, which must not start with a slash (e.g. `service/health`).
+* `initialDelayMilliseconds`: Number of milliseconds to wait until executing the first HTTP call
+* `retries`: Max number of HTTP call attempts that this will execute until giving up and returning an error
+* `retriesDelayMilliseconds`: Number of milliseconds to wait between retries
+* `bodyText`: If this value is non-empty, the endpoint will not be marked as available until this value is returned (e.g. `Hello World`). If this value is emptystring, no body text comparison will be done.
+
+### waitForEndpointAvailabilityHttpPost(ServiceID serviceId, uint32 port, String path, String requestBody, uint32 initialDelayMilliseconds, uint32 retries, uint32 retriesDelayMilliseconds, String bodyText)
+Waits until a service endpoint is available by making requests to the endpoint using the given parameters, and the http get method. An error is thrown if the number of retries is exceeded.
```
Comment (mieubrisse) [inline docs/lib-documentation.md:137]: s/http get/HTTP POST/g

File: docs/lib-documentation.md (hunk 1 above)
Comment (mieubrisse) [inline docs/lib-documentation.md:136]: Same thing, here and everywhere else in the PR - swapping for `waitForHttpPostEndpointAvailability` naming, which will be a little clearer

File: golang/lib/binding_constructors/binding_constructors.go [hunk 7]
```diff
@@ -232,25 +232,45 @@ func NewExecCommandResponse(exitCode int32, logOutput []byte) *kurtosis_core_rpc
 }
 
 // ==============================================================================================
-//                              Wait For Endpoint Availability
+//                           Wait For Endpoint Availability Http Get
 // ==============================================================================================
-func NewWaitForEndpointAvailabilityArgs(
+func NewWaitForEndpointAvailabilityHttpGetArgs(
```
Comment (mieubrisse) [inline golang/lib/binding_constructors/binding_constructors.go:237]: `NewWaitForHttpGetEndpointAvailabilityArgs` will be more natural here

File: golang/lib/binding_constructors/binding_constructors.go [hunk 8]
```diff
@@ -232,25 +232,45 @@ func NewExecCommandResponse(exitCode int32, logOutput []byte) *kurtosis_core_rpc
...
+		Retries:                  retries,
+		RetriesDelayMilliseconds: retriesDelayMilliseconds,
+		BodyText:                 bodyText,
+	}
+}
+
+// ==============================================================================================
+//                           Wait For Endpoint Availability Http Post
+// ==============================================================================================
+func NewWaitForEndpointAvailabilityHttpPostArgs(
```
Comment (mieubrisse) [inline golang/lib/binding_constructors/binding_constructors.go:259]: same naming swap here

File: golang/lib/bulk_command_execution/v0_bulk_command_api/v0_command_types.go [hunk 9]
```diff
@@ -32,26 +32,30 @@ type V0CommandTypeVisitor interface {
...
+	LoadLambdaCommandType                          V0CommandType = "LOAD_LAMBDA"
+	ExecuteLambdaCommandType                       V0CommandType = "EXECUTE_LAMBDA"
+	RegisterServiceCommandType                     V0CommandType = "REGISTER_SERVICE"
+	GenerateFilesCommandType                       V0CommandType = "GENERATE_FILES"
+	LoadStaticFilesCommandType                     V0CommandType = "LOAD_STATIC_FILES"
+	StartServiceCommandType                        V0CommandType = "START_SERVICE"
+	RemoveServiceCommandType                       V0CommandType = "REMOVE_SERVICE"
+	RepartitionCommandType                         V0CommandType = "REPARTITION"
+	ExecCommandCommandType                         V0CommandType = "EXEC_COMMAND"
+	WaitForEndpointAvailabilityHttpGetCommandType  V0CommandType = "WAIT_FOR_ENDPOINT_AVAILABILITY_HTTP_GET"
```
Comment (mieubrisse) [inline golang/lib/bulk_command_execution/v0_bulk_command_api/v0_command_types.go:53]: Same name swap, etc.

File: golang/lib/networks/network_context.go [hunk 10]
```diff
@@ -397,7 +395,34 @@ func (networkCtx *NetworkContext) WaitForEndpointAvailability(serviceId services
 			retriesDelayMilliseconds,
 		)
 	}
+	return nil
+}
+
+// Docs available at https://docs.kurtosistech.com/kurtosis-client/lib-documentation
+func (networkCtx *NetworkContext) WaitForEndpointAvailabilityHttpPost(serviceId services.ServiceID, port uint32, path string, requestBody string, initialDelayMilliseconds uint32, retries uint32, retriesDelayMilliseconds uint32, bodyText string) error {
```
Comment (mieubrisse) [inline golang/lib/networks/network_context.go:402]: Name swaps here

File: kurtosis-core-rpc-api/api_container_service.proto [hunk 11]
```diff
@@ -49,8 +49,11 @@ service ApiContainerService {
   // Executes the given command inside a running container
   rpc ExecCommand(ExecCommandArgs) returns (ExecCommandResponse) {};
 
-  // Block until the given HTTP endpoint returns available
-  rpc WaitForEndpointAvailability(WaitForEndpointAvailabilityArgs) returns (google.protobuf.Empty) {};
+  // Block until the given HTTP endpoint returns available, calling it throw a HTTP Get request
+  rpc WaitForEndpointAvailabilityHttpGet(WaitForEndpointAvailabilityHttpGetArgs) returns (google.protobuf.Empty) {};
+
+  // Block until the given HTTP endpoint returns available, calling it throw a HTTP Post request
```
Comment (mieubrisse) [inline kurtosis-core-rpc-api/api_container_service.proto:55]: s/throw/through/

File: scripts/regenerate-protobuf-output.sh [hunk 12]
```diff
@@ -24,7 +24,7 @@ fi
 echo "Successfully generated Go bindings in directory '${go_output_dirpath}'"
 
 # TypeScript
-typescript_output_dirpath="${root_dirpath}/${TYPESCRIPT_DIRNAME}/kurtosis_core_rpc_api_bindings"
+typescript_output_dirpath="${root_dirpath}/${TYPESCRIPT_DIRNAME}/src/kurtosis_core_rpc_api_bindings"
```
Comment (mieubrisse) [inline scripts/regenerate-protobuf-output.sh:27]: sorry... 😅

File: typescript/src/lib/bulk_command_execution/v0_bulk_command_api/v0_command_types.ts [hunk 13]
```diff
@@ -28,7 +29,8 @@ export enum V0CommandType {
 	RemoveService = "REMOVE_SERVICE",
 	Repartition = "REPARTITION",
 	ExecCommand = "EXEC_COMMAND",
-	WaitForEndpointAvailability = "WAIT_FOR_ENDPOINT_AVAILABILITY",
+	WaitForEndpointAvailabilityHttpGet = "WAIT_FOR_ENDPOINT_AVAILABILITY_HTTP_GET",
```
Comment (mieubrisse) [inline typescript/src/lib/bulk_command_execution/v0_bulk_command_api/v0_command_types.ts:32]: Make sure to propagate the renames to this guy

File: typescript/src/lib/constructor_calls.ts [hunk 14]
```diff
@@ -174,23 +174,41 @@ export function newPartitionConnections(allConnectionInfo: Map<string, Partition
     return result;
 }
 
-export function newWaitForEndpointAvailabilityArgs(
+export function newWaitForEndpointAvailabilityHttpGetArgs(
```
Comment (mieubrisse) [inline typescript/src/lib/constructor_calls.ts:177]: Nice job adding these constructors!! Make sure to just propagate the rename swap here

**Review by mieubrisse — APPROVED** [review]
Approved pending comments

File: docs/lib-documentation.md (hunk 3 above)
Comment (leoporoli) [reply docs/lib-documentation.md:131]: Yes, I did it in Kurt Core

File: docs/changelog.md (hunk 4 above)
Comment (leoporoli) [reply docs/changelog.md]: Done!

**Pull request merged**